src/opm/parser/eclipse/EclipseState/Schedule/UDQ/UDQFunctionTable.cpp

```diff
@@ -132,9 +133,19 @@ bool UDQFunctionTable::operator==(const UDQFunctionTable& data) const {
     if (this->functionMap().size() != data.functionMap().size())
         return false;
 
-    auto tIt = this->functionMap().begin();
-    auto dIt = data.functionMap().begin();
-    for (; tIt != this->functionMap().end(); ++tIt, ++dIt) {
+    // Simply using the operator== method of std::unordered_map would
+    // compare the pointers contained and not the objects pointed to.
+    // We therefore must iterate manually through the container. The
+    // order of the map elements does not matter, and is
+    // indeterminate. Sorting must be done before comparision so that
+    // we compare elements with the same key to each other.
+    using SortedMap = std::map<std::string, std::shared_ptr<UDQFunction>>;
+    SortedMap sorted_fmap(this->functionMap().begin(), this->functionMap().end());
+    SortedMap data_sorted_fmap(data.functionMap().begin(), data.functionMap().end());
```

Comment:
I would prefer:
```
if (this->functionMap().size() != data.functionMap().size())
     return false;

for (const auto& fpair : this->functionMap()) {
      if (data.functionMap().count(fpair.first) != 1)
          return false;

....

}
```
but I'll merge yours if you prefer that.

Reply:
I do not see any comparison between the elements here? So for that reason I prefer mine... If we do it with manual lookup we should get the iterator, right?

Reply:
> I do not see any comparison between the elements here

That is the dots ...

> If we do it with manual lookup we should get the iterator, right?

??

Reply:
Already merged, so moot point... What I mean is that we should use find() to get the iterators into the two containers, and then use those for all checking rather than doing first a lookup with count() and then another lookup to check the element equality.

For later: the reason for you preferring to avoid the std::map here, is that to (hopefully) get O(n) behaviour rather than O(n log n)?